installinator/src/write.rs

```diff
@@ -777,6 +806,88 @@ impl ControlPlaneZoneWriteContext<'_> {
                 .register();
         }
 
+        engine
+            .new_step(
+                WriteComponent::ControlPlane,
+                ControlPlaneZonesStepId::CreateMeasurementDir,
+                "Creating measurement directory".to_string(),
+                async move |_cx| {
+                    if !std::fs::exists(self.measurement_directory)
+                        .map_err(|error| WriteError::CreateDirError { error })?
+                    {
+                        std::fs::create_dir(self.measurement_directory)
+                            .map_err(|error| WriteError::CreateDirError {
+                                error,
+                            })?;
+                    }
+                    StepSuccess::new(()).into()
+                },
+            )
+            .register();
+
+        transport = engine
+            .new_step(
+                WriteComponent::ControlPlane,
+                ControlPlaneZonesStepId::ZoneManifest,
+                "Writing measurement manifest",
+                async move |cx| {
+                    let transport = transport.into_value(cx.token()).await;
+                    let measurement_manifest_json =
+                        self.omicron_measurement_manifest_artifact().await;
+
+                    let out_path = self
+                        .measurement_directory
+                        .join(OmicronZoneManifest::MEASUREMENT_FILE_NAME);
+
+                    write_artifact_impl(
+                        WriteComponent::MeasurementCorpus,
+                        slot,
+                        measurement_manifest_json,
+                        &out_path,
+                        transport,
+                        &cx,
+                    )
+                    .await?;
+
+                    StepSuccess::new(transport)
+                        .with_message(format!(
+                            "{out_path} written with mupdate UUID: {}",
+                            self.mupdate_id,
+                        ))
+                        .into()
+                },
+            )
+            .register();
+
+        // This might make sense to move to a separate step once reconfigurator
+        // work is complete. For now, just treat this as another step in the
+        // control plane
```

Comment:
Not sure how to interpret this comment - What does reconfigurator work have to do with installinator's steps?